### PR TITLE
Toggle remove option for string definition of instruction (#2059)

### LINKED ISSUE #2059: Don't show "Set as... -> String... -> Remove" where no string is defined
**Environment information**
* Operating System: Any
* Cutter version: Recent from Git
* Obtained from:
  - [x] Built from source
* File format: Any

**Describe the bug**

Currently, when right clicking a non-string instruction, there is the option to remove the definition of the instruction as string while it is actually a code. This shouldn't be available unless it is indeed defined as string/

**To Reproduce**

Right click on an instruction and see that "Set as... -> String... -> Remove" is available. 
image

**Expected behavior**

The option to remove the definition of a String should only be there when the instruction is defined as string.

### PR DESCRIPTION
**Pull request by dhaval0894**
**Things to be discussed:**
- The way the functionality is implemented.

**Detailed description**

**Functional Changes**
- The remove option would be visible once the String definition is set.
- The remove option would be hidden once the String definition is removed.

**Code Changes**
In **DisassemblyContextMenu.cpp**, following changes are added:
- Seek to instruction offset `seek(offset)`.
- Retrieve string definition at the address using `Cs.`. 
- If the string definition is present, set the visibility of `actionSetAsStringRemove` to true, otherwise, set it to false.

**Problem solved by pull request**
- The option to remove the definition of a String will only be there when the instruction is defined as String.

**Test plan (required)**

**Changes in the UI**
- For instruction at address `0x000021c4`, we verify that the remove option is hidden when the string definition is not set:
test-1

- Once the String definition is set for instruction at `0x000021c4`, we check the remove option for another instruction at address `0x000021e1` and we make sure that it is not visible in the menu: 
test-2

- Finally, for the instruction at `0x000021c4`, we ensure that the remove option is now visible in the menu as the string definition is set:
test-3

**Closing issues**

Closes #2059

### CODE REVIEW

**Comment by dhaval0894** [comment]
Hi @karliss! I have changed the implementation as per the procedure you suggested and also updated the test plan. Please have a look.

Thank you for describing the process, it really helped a lot and I would keep this in mind for the next time. I kept digging at the Cutter layer and ended up wasting a lot of time. Going a layer deeper at the r2 level made a lot of things clear. 

Also, thanks a lot @ITAYC0HEN for showing me the right way and helping me out with the understanding of the r2 commands. Your [tutorial](https://www.megabeets.net/a-journey-into-radare-2-part-1/) is absolutely amazing! It provided a pretty solid overview of radare2 and consequently, it helped me to understand the implementation:).

I knew that my previous implementation had problems but my idea behind creating the PR was to quickly understand whether I have taken the correct approach or not. Although, now the solution seems pretty straightforward and I should have finished it in a reasonable amount of time but nevertheless, the whole process helped me understand a lot of things.

I would try to be more competent with the next one and at the same time, would try to finish it in a reasonable time.

File: src/menus/DisassemblyContextMenu.cpp [hunk 1]
```diff
@@ -459,6 +459,14 @@ void DisassemblyContextMenu::aboutToShowSlot()
 
     actionAnalyzeFunction.setVisible(true);
 
+    Core()->seek(offset);
+    QString stringDefiniton = Core()->cmd("Cs.");
+    if (stringDefiniton.isNull() || stringDefiniton.isEmpty()) {
+        actionSetAsStringRemove.setVisible(false);
+    } else {
+        actionSetAsStringRemove.setVisible(true);
+    }
```
Comment (ITAYC0HEN) [inline src/menus/DisassemblyContextMenu.cpp]:
```suggestion
    actionSetAsStringRemove.setVisible(!stringDefiniton.isEmpty());
```

File: src/menus/DisassemblyContextMenu.cpp [hunk 2]
```diff
@@ -459,6 +459,14 @@ void DisassemblyContextMenu::aboutToShowSlot()
 
     actionAnalyzeFunction.setVisible(true);
 
+    Core()->seek(offset);
+    QString stringDefiniton = Core()->cmd("Cs.");
```
Comment (ITAYC0HEN) [inline src/menus/DisassemblyContextMenu.cpp]: Please add a comment above this line saying: "// Show the option to remove a defined string only if a string is defined in this address"

**Comment by dhaval0894** [comment]
Hi, I have made the requested changes.

**Pull request merged**